src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginDiscoverer.cs

```diff
@@ -183,7 +182,7 @@ internal class TestPluginDiscoverer
         /// <typeparam name="TExtension">
         /// Type of Extensions.
         /// </typeparam>
-        private void GetTestExtensionsFromAssembly<TPluginInfo, TExtension>(Assembly assembly, Dictionary<string, TPluginInfo> pluginInfos) where TPluginInfo : TestPluginInformation
+        internal virtual void GetTestExtensionsFromAssembly<TPluginInfo, TExtension>(Assembly assembly, Dictionary<string, TPluginInfo> pluginInfos) where TPluginInfo : TestPluginInformation
```

Comment:
Is this for unit tests? Can we drive the test scenario through a public method?

(looks like a design issue in existing code)